datatable/java/datatable/src/main/java/io/cucumber/datatable/DataTableTypeRegistry.java

```diff
@@ -57,6 +58,44 @@ public DataTableTypeRegistry(Locale locale) {
         TableCellTransformer<Double> doubleTableCellTransformer = applyIfPresent(numberParser::parseDouble);
         defineDataTableType(new DataTableType(Double.class, doubleTableCellTransformer));
         defineDataTableType(new DataTableType(double.class, doubleTableCellTransformer));
+
+        TableCellTransformer<Optional<Object>> optionalTableCellTransformer =
+                (String s) -> s == null || s.isEmpty() ? Optional.empty() : Optional.of(s);
+        defineDataTableType(new DataTableType(Optional.class, optionalTableCellTransformer));
+
+        Type optionalString = new TypeReference<Optional<String>>() { }.getType();
+        defineDataTableType(new DataTableType(optionalString, optionalTableCellTransformer));
+
+        Type optionalDouble = new TypeReference<Optional<Double>>() { }.getType();
+        TableCellTransformer<Optional<Double>> optionalDoubleTableCellTransformer =
+                (String s) -> s == null || s.isEmpty() ? Optional.empty() : Optional.of(numberParser.parseDouble(s));
+        defineDataTableType(new DataTableType(optionalDouble, optionalDoubleTableCellTransformer));
+
+        Type optionalFloat = new TypeReference<Optional<Float>>() { }.getType();
+        TableCellTransformer<Optional<Float>> optionalFloatTableCellTransformer =
+                (String s) -> s == null || s.isEmpty() ? Optional.empty() : Optional.of(numberParser.parseFloat(s));
```

Comment:
The transformer specific function (e.g. `numberParser.parseFloat`) for both the optional and non-optional transformer should be the same. However it is very hard to verify that this is the case. By moving the definition of the optional and non-optional transformer closer together you can resolve this.

Additionally, in combination with my previous comment you can possibly also make it easier to verify that the transformer specific part is indeed the same.